internal/server/handlers/validation/pod_scheduling_policy.go

```diff
@@ -0,0 +1,223 @@
+	switch psp.Spec.EngineType {
+	case everestv1alpha1.DatabaseEnginePXC:
+		if affinityConfig.PSMDB != nil {
+			// .spec.affinityConfig.psmdb shall not be set with engineType=pxc
+			return fmt.Errorf("invalid affinity config: .spec.affinityConfig.psmdb is not applicable with engineType='%s'", everestv1alpha1.DatabaseEnginePXC)
+		}
+		if affinityConfig.PostgreSQL != nil {
+			// .spec.affinityConfig.postgresql shall not be set with engineType=pxc
+			return fmt.Errorf("invalid affinity config: .spec.affinityConfig.postgresql is not applicable with engineType='%s'", everestv1alpha1.DatabaseEnginePXC)
+		}
+		if affinityConfig.PXC == nil {
+			return errors.New("invalid affinity config: .spec.affinityConfig.pxc is required")
+		}
+		if affinityConfig.PXC.Engine == nil && affinityConfig.PXC.Proxy == nil {
+			return errors.New("invalid affinity config: .spec.affinityConfig.pxc.engine or .spec.affinityConfig.pxc.proxy is required")
+		}
+	case everestv1alpha1.DatabaseEnginePSMDB:
+		if affinityConfig.PXC != nil {
+			// .spec.affinityConfig.pxc shall not be set with engineType=psmdb
+			return fmt.Errorf("invalid affinity config: .spec.affinityConfig.pxc is not applicable with engineType='%s'", everestv1alpha1.DatabaseEnginePSMDB)
+		}
+		if affinityConfig.PostgreSQL != nil {
+			// .spec.affinityConfig.postgresql shall not be set with engineType=psmdb
+			return fmt.Errorf("invalid affinity config: .spec.affinityConfig.postgresql is not applicable with engineType='%s'", everestv1alpha1.DatabaseEnginePSMDB)
+		}
+		if affinityConfig.PSMDB == nil {
+			return errors.New("invalid affinity config: .spec.affinityConfig.psmdb is required")
+		}
+		if affinityConfig.PSMDB.Engine == nil && affinityConfig.PSMDB.Proxy == nil && affinityConfig.PSMDB.ConfigServer == nil {
+			return errors.New("invalid affinity config: .spec.affinityConfig.psmdb.engine or .spec.affinityConfig.psmdb.proxy or .spec.affinityConfig.psmdb.configServer is required")
+		}
+	case everestv1alpha1.DatabaseEnginePostgresql:
+		if affinityConfig.PXC != nil {
+			// .spec.affinityConfig.pxc shall not be set with engineType=postgresql
+			return fmt.Errorf("invalid affinity config: .spec.affinityConfig.pxc is not applicable with engineType='%s'", everestv1alpha1.DatabaseEnginePostgresql)
+		}
+		if affinityConfig.PSMDB != nil {
+			// .spec.affinityConfig.psmdb shall not be set with engineType=postgresql
+			return fmt.Errorf("invalid affinity config: .spec.affinityConfig.psmdb is not applicable with engineType='%s'", everestv1alpha1.DatabaseEnginePostgresql)
+		}
+		if affinityConfig.PostgreSQL == nil {
+			return errors.New("invalid affinity config: .spec.affinityConfig.postgresql is required")
+		}
+		if affinityConfig.PostgreSQL.Engine == nil && affinityConfig.PostgreSQL.Proxy == nil {
+			return errors.New("invalid affinity config: .spec.affinityConfig.postgresql.engine or .spec.affinityConfig.postgresql.proxy is required")
+		}
+	}
+	return nil
+}
```

Comment:
since we have similar error messages everywhere, suggestion to create a dedicated error type and use that:

```go
type pspValidationError struct {
	err error
}

func (e *pspValidationError) Error() string {
	return fmt.Sprintf("invalid affinity config: %v", e.err)
}

func newPspValidationError(msg string) *pspValidationError {
	return &pspValidationError{
		err: fmt.Errorf(msg),
	}
}

```